source/Planet.cpp

```diff
@@ -227,7 +237,7 @@ void Planet::Load(const DataNode &node, Set<Wormhole> &wormholes)
 	}
 
 	static const vector<string> AUTO_ATTRIBUTES = {"spaceport", "shipyard", "outfitter"};
-	bool autoValues[3] = {!spaceport.empty(), !shipSales.empty(), !outfitSales.empty()};
+	bool autoValues[3] = {port.HasService(Port::ServicesType::All), !shipSales.empty(), !outfitSales.empty()};
```

Comment:
Since a port can now have different services, I think we should have autoattributes for each service. We'd keep the existing "spaceport" attribute as you're doing now to represent a port with all services, but then we'd have individual "bank", "trade", etc. attributes for each service a port could have.

Reply:
Well I said this same thing in 2021 but said it could be a different PR... I want it in this PR now. :smug: